src/Microsoft.OData.Core/JsonLight/ODataJsonLightDeltaWriter.cs

```diff
@@ -287,6 +287,12 @@ void IODataOutputInStreamErrorListener.OnInStreamError()
             this.inStreamErrorListener.OnInStreamError();
         }
 
+        /// <inheritdoc/>
+        Task IODataOutputInStreamErrorListener.OnInStreamErrorAsync()

```

Comment:
Out of curiosity, why is this method explicitly defined for the `IODataOutputInStreamErrorListener` interface? I realize that this is also the case for the pre-existing `OnInStreamError` methods, but just curious to know why. Does this class implement multiple interfaces each defining `OnInStreamError`?

Reply:
> Does this class implement multiple interfaces each defining OnInStreamError?
No it doesn't
> why is this method explicitly defined for the IODataOutputInStreamErrorListener interface?
I'm not 100% sure why explicit interface implementation route was taken

Reply:
I'm guessing the benefit  is that any class that implements `ODataBatchWriter` abstract class for example can override the default `IODataOutputInStreamErrorListener` methods either as a public method, or as an explicit interface implementation.

Reply:
@habbes A person who implements `ODataBatchWriter` abstract class would only bring it into play by implementing the `public abstract` **`ODataOutputContext`** `CreateODataBatchWriter` method just like we do in [`ODataJsonLightOutputContext`](https://github.com/OData/odata.net/blob/de68595d34fa77ab2b192b1d7c6d35682a4100dd/src/Microsoft.OData.Core/JsonLight/ODataJsonLightOutputContext.cs#L564).
By doing that they take the internal `ODataJsonLightOutputContext` out of play and take the responsibility of triggering the `IODataOutputInStreamErrorListener` methods.
`ODataJsonLightOutputContext` has a hard dependency of `ODataJsonLightBatchWriter` as you can see [here](https://github.com/OData/odata.net/blob/de68595d34fa77ab2b192b1d7c6d35682a4100dd/src/Microsoft.OData.Core/JsonLight/ODataJsonLightOutputContext.cs#L931)
```csharp
private ODataBatchWriter CreateODataBatchWriterImplementation()
{
    ODataBatchWriter batchWriter = new ODataJsonLightBatchWriter(this);
    this.outputInStreamErrorListener = batchWriter;
    return batchWriter;
}
```